internal/pkg/v2/infrastructure/redis/reading.go

```diff
@@ -231,15 +231,29 @@ func readingsByTimeRange(conn redis.Conn, start int, end int, offset int, limit
 
 func convertObjectsToReadings(objects [][]byte) (readings []models.Reading, edgeXerr errors.EdgeX) {
 	readings = make([]models.Reading, len(objects))
+	var alias struct {
+		ValueType string
+	}
 	for i, in := range objects {
-		// as V2 APi doesn't deal with BinaryReading at this moment, convert to SimpleReading here
-		// Shall update the logic here when working on BinaryReading in the future
-		sr := models.SimpleReading{}
-		err := json.Unmarshal(in, &sr)
+		err := json.Unmarshal(in, &alias)
 		if err != nil {
 			return []models.Reading{}, errors.NewCommonEdgeX(errors.KindDatabaseError, "reading format parsing failed from the database", err)
 		}
-		readings[i] = sr
+		if alias.ValueType == v2.ValueTypeBinary {
+			var binaryReading models.BinaryReading
+			err = json.Unmarshal(in, &binaryReading)
+			if err != nil {
+				return []models.Reading{}, errors.NewCommonEdgeX(errors.KindDatabaseError, "binary reading format parsing failed from the database", err)
+			}
+			readings[i] = binaryReading
```

Comment:
the `binaryValue` field should be assigned an empty value because we don't want to persist the binaryValue

Reply:
It's empty when we read the data from the RedisDB because we already assign the empty value when storing the reading.
https://github.com/edgexfoundry/edgex-go/blob/master/internal/pkg/v2/infrastructure/redis/reading.go#L99